src/main/java/net/mcreator/element/types/Item.java

```diff
@@ -220,14 +223,105 @@ public List<StateEntry> getModels() {
 					model.stateMap.put(prop, value);
 			});
 
-			// only add this state if at least one supported property is present
-			if (!model.stateMap.isEmpty())
+			// only add this state if not duplicate and at least one supported property is present
+			if (!models.contains(model) && !model.stateMap.isEmpty()) {
+				model.listIndex = index.getAndIncrement();
 				models.add(model);
+			}
 		});
 		return models;
 	}
 
-	public static class StateEntry implements IWorkspaceDependent {
+	public StateEntry getBaseState() {
+		StateEntry baseState = new StateEntry();
+		baseState.renderType = renderType;
+		baseState.texture = texture;
+		baseState.customModelName = customModelName;
+		return baseState;
+	}
+
+	/**
+	 * Returns a tree-like states representation referencing only properties supported in the current workspace.
+	 * For use by 1.21.4+ generators.
+	 *
+	 * @return Models tree with contents matching current generator.
+	 */
+	public StateTreeNode getModelsTree() {
+		StateEntry baseState = getBaseState();
+
+		// initially set to item's default state
+		// if no model overrides are defined, this will return tree of depth 1
+		StateTreeNode root = baseState;
+		for (StateEntry state : getModels()) {
+			if (state.stateMap.isEmpty())
+				continue; // should never happen because states with empty StateMap are removed by getModels()
+
+			StateBranch current = new StateBranch(null, root);

```

Comment:
This code would need some more comments on what exactly it does

Reply:
`root` is not guaranteed to be a branch, so we have to make one for it. How could this be explained?